tests/unit/references/test_processors.py

```diff
@@ -0,0 +1,120 @@
+
+def test_reference_builder_no_uids():
+    rb = ReferenceBuilder()
+    rb.set_number('oops')
+    rb.set_number('1')
+    rb.set_texkey('book')
+    rb.add_title('Awesome Paper')
+    rb.add_raw_reference('[1] Awesome Paper')
+    rb.add_misc('misc 0')
+    rb.add_misc('misc 1')
+    rb.add_author('Cox, Brian')
+    rb.add_author('O Briain, Dara', 'ed.')
+    rb.set_pubnote('Nucl.Phys.,B360,362')
+    rb.set_pubnote('BAD PUBNOTE')
+    rb.set_year('oops')
+    rb.set_year('1991')
+    rb.add_url('http://example.com')
+
+    rb.set_publisher('Elsevier')
+    rb.add_collaboration('ALICE')
+
+    rb.add_report_number('hep-th/0603001')
+    rb.add_report_number('hep-th/0603002')
+    rb.add_report_number('arXiv:0705.0016 [hep-th]')
+    rb.add_report_number('0705.0017 [hep-th]')
+    rb.add_report_number('NOT ARXIV')
+
+    expected = {
+        'number': 1,
+        'texkey': 'book',
+        'titles': [{'value': 'Awesome Paper'}],
+        'raw_reference': [
+            {'value': '[1] Awesome Paper', 'serialization': 'text'},
+            {'value': 'BAD PUBNOTE', 'serialization': 'text'}
+        ],
+        'misc': ['misc 0', 'misc 1'],
+        'authors': [{'full_name': 'Cox, Brian'},
+                    {'full_name': 'O Briain, Dara', 'role': 'ed.'}],
+        'publication_info': {
+            'journal_title': 'Nucl.Phys.',
+            'journal_volume': 'B360',
+            'page_start': '362',
+            'artid': '362',
+            'reportnumber': 'NOT ARXIV',
+            'year': 1991
+        },
+        'collaboration': ['ALICE'],
+        'imprint': {'publisher': 'Elsevier'},
+        'arxiv_eprints': ['arXiv:hep-th/0603001', 'arXiv:hep-th/0603002',
+                          'arXiv:0705.0016', 'arXiv:0705.0017'],
+        'urls': [{'value': 'http://example.com'}]
+    }
+
+    assert expected == rb.obj
+
+
+def test_curation():
+    rb = ReferenceBuilder()
+    rb.set_record({'$ref': 'http://example.com'})
+
+    assert rb.obj == {'record': {'$ref': 'http://example.com'},
+                      'curated_relation': False}
+
+    rb.curate()
+    assert rb.obj == {'record': {'$ref': 'http://example.com'},
+                      'curated_relation': True}
+
+
+def test_reference_builder_add_uid():
+    rb = ReferenceBuilder()
+    rb.add_uid(None)
+    rb.add_uid('')
+    rb.add_uid('thisisarandomstring')
+
+    # arXiv eprint variations
+    rb.add_uid('hep-th/0603001')
+    rb.add_uid('0705.0017 [something-th]')
+    # isbn
+    rb.add_uid('1449344852')
+    # cnum
+    rb.add_uid('C87-11-11')
+    # doi
+    rb.add_uid('http://dx.doi.org/10.3972/water973.0145.db')
+    # handle
+    rb.add_uid('hdl:10443/1646')
+
+    expected = {
+        'arxiv_eprints': ['arXiv:hep-th/0603001', 'arXiv:0705.0017'],
+        'publication_info': {
+            'isbn': '978-1-4493-4485-6',
+            'cnum': 'C87-11-11',
+            'doi': '10.3972/water973.0145.db'
+        },
+        'persistent_identifiers': ['10443/1646']
```

Comment:
hmm. I'd just add prepend `hdl:` in front of them when normalizing. maybe we'll change them together with the schema in #1279